### DISCUSSION TITLE
enable/disable TextInput depending on another select field's value

### DISCUSSION THREAD

**Discussion by OvalMedia** [filamentphp/filament#4697] [Help]
I have a dropdown and a text input. I would like the text input to be enabled/disabled dependant on wether the select has any value.
From the docs I know I can get and set the **value** of another field in a callback function. But how can I modify any other aspect of other fields? Like in this case it's disabled state?

This is what I tried but this way the input is only disabled after load. It does not change when I select an option.
```

Select::make('unit')
	->reactive()
	->options([
		'hour' => 'Hour(s)',
		'day' => 'Day(s)'
	]),
TextInput::make('rate')	
	->disabled(fn (Closure $get) => $get('unit') == '')

```

**Comment by danharrin** [comment]
Probably because `unit` is `''` not `null`... try replacing `is_null` with `blank`

**Reply by OvalMedia** [reply]
I updated the snippet (`$get('unit') == ''`) but it does not make a difference. 
The rate field is enabled or disabled only after load. So the general evaluation works, it is just not being reevaluated after the select is updated.

**Reply by danharrin** [reply]
As I said, try `blank()` instead of `is_null()`. This handles both `null` and `''`.

```php
TextInput::make('rate')	
    ->disabled(fn (Closure $get): bool => blank($get('unit')))
```

**Reply by OvalMedia** [reply]
Still nothing.

**Reply by danharrin** [reply]
I just tried it and it does work.

https://user-images.githubusercontent.com/41773797/197492734-828e81a3-af81-4272-9961-a06c04c058e6.mov

**Reply by OvalMedia** [reply]
Alright, thx for your effort. For the sake of abbreviation I had left one part out which seemed unrelated.
The rate field has a money mask (`->mask(fn ($mask) => $mask->money('€'))`). When I remove this part the enabling/disabling actually works.

**Reply by danharrin** [reply]
Try `->prefix('€')` instead
